tests/utils/test_parallelize.py

```diff
@@ -0,0 +1,104 @@
+"""Tests for verifying process/thread usage in parallelized functions."""
+
+from __future__ import annotations
+
+import time
+from collections.abc import Callable
+from functools import partial
+
+import dask.array as da
+import numba
+import numpy as np
+import psutil
+import pytest  # type: ignore[import]
+
+from squidpy._utils import Signal, parallelize
+
+# Functions to be parallelized
+
+
+@numba.njit(parallel=True)
+def numba_parallel_func(x, y) -> np.ndarray:
+    return x * 2 + y
+
+
+@numba.njit(parallel=False)
+def numba_serial_func(x, y) -> np.ndarray:
+    return x * 2 + y
+
+
+def dask_func(x, y) -> np.ndarray:
+    return (da.from_array(x) * 2 + y).compute()
+
+
+def vanilla_func(x, y) -> np.ndarray:
+    return x * 2 + y
+
+
+# Mock runner function
+
+
+def mock_runner(x, y, queue, func):
+    for i in range(len(x)):
+        x[i] = func(x[i], y)
+        if queue is not None:
+            queue.put(Signal.UPDATE)
+    if queue is not None:
+        queue.put(Signal.FINISH)
+    return x
+
+
+@pytest.fixture(params=["numba_parallel", "numba_serial", "dask", "vanilla"])
+def func(request) -> Callable:
+    return {
+        "numba_parallel": numba_parallel_func,
+        "numba_serial": numba_serial_func,
+        "dask": dask_func,
+        "vanilla": vanilla_func,
+    }[request.param]
+
+
+@pytest.mark.timeout(60)
+@pytest.mark.parametrize("n_jobs", [1, 2, 8])
+def test_parallelize_loky(func, n_jobs):
+    start_time = time.time()
+    seed = 42
+    rng = np.random.RandomState(seed)
+    n = 8
+    arr1 = [rng.randint(0, 100, n) for _ in range(n)]
+    arr2 = np.arange(n)
+    runner = partial(mock_runner, func=func)
+    # this is the expected result of the function
+    expected = [func(arr1[i], arr2) for i in range(len(arr1))]
+    # this will be set to something other than 1,2,8
+    # we want to check if setting the threads works
+    # then after the function is run if the numba cores are set back to 1
+    old_num_threads = 3
+    numba.set_num_threads(old_num_threads)
+    # Get initial state
+    initial_process = psutil.Process()
+    initial_children = {p.pid for p in initial_process.children(recursive=True)}
+    initial_children = {psutil.Process(pid) for pid in initial_children}
```

Comment:
```python
initial_process = psutil.Process()
initial_children = {p.pid for p in initial_process.children(recursive=True)}
```
why would you want to make these `Process` objects, especially if they are set-diffed below with `pid` (which is a number)?

Reply:
I have a suspicion this is an `xdist` issue - I've run an individual test now 12 times and counting without failure where I added
```python
    assert len(initial_children) == 0
```
early in the test.

So I think we should look into ensuring all of these tests run in a single group (see slack), which could help with determinism.  And if we still can't get it working, it would _definitely_ be worth commenting (or investigating hooking into `xdist` and checking process numbers from its workers).  I think there is a way to check this via an api.  It's possible this issue of children processes only occurs on the master process

Reply:
Prinitng out the process ids, I can see that when not run in parallel, the first test has no children i.e., `numba_parallel-1` and then all subsequent runs have a single child process which indicates to me that there is something actually going on here with resource clean up.

You can see this by changin `n_jobs` to `[2,1,8]` - the first test has two children (makes sense), but then subsequent tests all have the same 4 children, two of which are the first two children! I think something is not being cleaned up.